test/base/index.js

```diff
@@ -0,0 +1,133 @@
+
+    assert.equal(customValidation.syncValidations.length, 2, 'two sync validations');
+    assert.equal(customValidation.customValidation, customValidationProps.validate, 'with custom validation');
+  });
+
+  it('getValidation', () => {
+    const props = {
+      required: true,
+      pattern: '[0-9]*',
+      validate: (value) => value.length === 5,
+    };
+
+    const instance = shallow(React.createElement(Child, props)).instance();
+
+    return Promise.all([
+      assert.isRejected(instance.getValidation(), null, null, 'no value'),
+      assert.isRejected(instance.getValidation('aaaaaa'), null, null, 'no pattern match'),
+      assert.isRejected(instance.getValidation('123'), null, null, 'validate not passed'),
+      assert.isFulfilled(instance.getValidation('12345'), null, null, 'correct value'),
+    ]);
+  });
+
+  it('setValue', () => {
+    const props = {
+      onUpdate: fake(),
+    };
+
+    const doneFake = fake();
+    const instance = shallow(React.createElement(Child, props)).instance();
+
+    instance.setValue('value', doneFake);
+    assert.equal(instance.getValue(), 'value', 'value was updated');
+    assert.equal(props.onUpdate.callCount, 1, 'on update was called');
+    assert.equal(doneFake.callCount, 1, 'callback was called');
+
+    instance.setValue('new value', doneFake, { silent: true });
+    assert.equal(instance.getValue(), 'new value', 'value was updated');
+    assert.equal(props.onUpdate.callCount, 1, 'on update was not called');
+    assert.equal(doneFake.callCount, 2, 'callback was called');
+  });
+
+  it('setError', () => {
+    const props = {
+      onError: fake(),
+    };
+
+    const doneFake = fake();
+    const instance = shallow(React.createElement(Child, props)).instance();
+
+    instance.setError('Error', doneFake);
+    assert.equal(instance.getError(), 'Error', 'error was set');
+    assert.equal(props.onError.callCount, 1, 'on error was called');
+    assert.equal(doneFake.callCount, 1, 'callback was called');
+  });
+
+  it('actionChange', () => {
+    const callback = fake();
+    const instance = shallow(React.createElement(Child)).instance();
+    const action = instance.actionChange(callback);
+
+    action({ target: { value: 'value' } });
+
+    assert.equal(instance.getValue(), 'value', 'value was updated');
+    assert.equal(callback.callCount, 1, 'callback was called');
+  });
+
+  it('actionClearError', () => {
+    const callback = fake();
+    const instance = shallow(React.createElement(Child)).instance();
+    const action = instance.actionClearError(callback);
+
+    action();
+
+    assert.equal(instance.getError(), null, 'error was cleared');
+    assert.equal(callback.callCount, 1, 'callback was called');
+  });
+
+  it('actionValidate', (done) => {
+    const callback = fake();
+    const props = { required: true, error: 'Error' };
+    const instance = shallow(React.createElement(Child, props)).instance();
+    const action = instance.actionValidate(callback);
```

Comment:
could probably update actionValicate to return promise - easier to test then. ok as it is too though

Reply:
let's change when there is a case